source/monitoring.txt

```diff
@@ -17,263 +17,415 @@ JMX Monitoring
+   * - ``port``
+     - The port on which the ``mongod`` or ``mongos`` process is listening.
+
+   * - ``minSize``
+     - The minimum size of the connection pool, including idle and in-use connections.
+
+   * - ``maxSize``
+     - The maximum size of the connection pool, including idle and in-use connections.
+
+   * - ``size``
+     - The current size of the connection pool, including idle and in-use connections.
+
+   * - ``checkedOutCount``
+     - The current count of connections that are in use.
+
+
+All MXBean instances created by the driver are under the domain
+``"org.mongodb.driver"``.
+
+For more information about the topics discussed in this subsection, see the
+following resources from Oracle:
+
+- `Platform MBean Server Reference Documentation <https://docs.oracle.com/en/java/javase/16/management/overview-java-se-monitoring-and-management.html#GUID-F7B9AB8A-F5A8-472A-AEC6-93B5B7FBE7CE>`__
+- `MXBean Documentation <https://docs.oracle.com/javase/tutorial/jmx/mbeans/mxbeans.html>`__
+- `MBean Documentation <https://docs.oracle.com/javase/tutorial/jmx/mbeans/standard.html>`__
+
+JMX and JConsole Example
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example shows how you can monitor the driver's connection pools using JMX
+and **JConsole**. JConsole is a JMX-compliant GUI monitoring tool that comes with
+the Java Platform.
+
+.. tip:: Consult the Official JMX and JConsole Documentation
+
+   The descriptions of JMX and JConsole in this example are illustrative
+   rather than a source of truth. For guaranteed up to date information, consult
+   the following official Oracle resources:
+
+   - `JConsole documentation <https://www.oracle.com/technical-resources/articles/java/jconsole.html>`__
+   - `JMX documentation <https://docs.oracle.com/javase/tutorial/jmx/index.html>`__
+
+The following code snippet adds a ``JMXConnectionPoolListener`` to a
+``MongoClient`` instance. The code then pauses execution so you can
+navigate to JConsole and inspect your connection pools.
+
+.. io-code-block::
+   :copyable: true
+
+   .. input:: /includes/monitoring/JMXMonitoring.java
+      :language: java
+      :start-after: start-jmx-example
+      :end-before: end-jmx-example
+      :dedent:
+
+   .. output::
+      :visible: false
+
+      Navigate to JConsole to see your connection pools...
+
+Once you have started your server, open JConsole by running the following command in your
+terminal:
+
+.. code-block:: shell
+
+   jconsole
+
+Once JConsole is open, perform the following actions in the GUI:
+
+#. Select the Java process running the preceding example code.
+#. Press :guilabel:`Insecure Connection` in the warning dialog box.
+#. Click on the :guilabel:`MBeans` tab.
+#. Inspect your connection pool events under the ``"org.mongodb.driver"`` domain.
+
+When you no longer want to inspect your connection pools in JConsole, do the
+following:
+
+- Exit JConsole by closing the JConsole window
+- Stop the Java program running the preceding code snippet
```

Comment:
[q] Do these need to be completed in order or it doesn't matter? Also I think can use punctuation as imperative statements.
```suggestion
- Exit JConsole by closing the JConsole window.
- Stop the Java program running the preceding code snippet.
```